lib/libzfs/os/linux/libzfs_mount_os.c

```diff
@@ -291,16 +551,31 @@ is_shared_impl(libzfs_handle_t *hdl, const char *mountpoint,
  * in the case of a mount failure we do not have the exact errno.  We must
  * make due with return value from the mount process.
  *
- * In the long term a shared library called libmount is under development
- * which provides a common API to address the locking and errno issues.
- * Once the standard mount utility has been updated to use this library
- * we can add an autoconf check to conditionally use it.
- *
- * http://www.kernel.org/pub/linux/utils/util-linux/libmount-docs/index.html
  */
 int
-do_mount(const char *src, const char *mntpt, char *opts, int flags)
+do_mount(zfs_handle_t *zhp, const char *mntpt, char *opts, int flags)
 {
+	const char *src = zfs_get_name(zhp);
+	if (! libzfs_envvar_is_set("ZFS_MOUNT_HELPER")) {
+		char badopt[MNT_LINE_MAX] = {0}, mtabopt[MNT_LINE_MAX] = {};
+		unsigned long mntflags = flags, zfsflags;
+		char *copy = strdup(opts);
```

Comment:
It doesn't look like you need to make a copy here since `zfs_parse_mount_options()` already does so internally.